pydicom/pixel_data_handlers/pylibjpeg_handler.py

```diff
@@ -0,0 +1,279 @@
++-------------+---------------------------+------+---------------+----------+
+| (0028,0010) | Rows                      | 1    | N             | Required |
++-------------+---------------------------+------+---------------+----------+
+| (0028,0011) | Columns                   | 1    | N             | Required |
++-------------+---------------------------+------+---------------+----------+
+| (0028,0100) | BitsAllocated             | 1    | 8, 16         | Required |
++-------------+---------------------------+------+---------------+----------+
+| (0028,0101) | BitsStored                | 1    | Up to 16      | Required |
++-------------+---------------------------+------+---------------+----------+
+| (0028,0103) | PixelRepresentation       | 1    | 0, 1          | Required |
++-------------+---------------------------+------+---------------+----------+
+
+"""
+
+import logging
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from pydicom.dataset import Dataset
+
+try:
+    import numpy as np
+    HAVE_NP = True
+except ImportError:
+    HAVE_NP = False
+
+try:
+    import pylibjpeg
+    from pylibjpeg.pydicom.utils import get_pixel_data_decoders
+    HAVE_PYLIBJPEG = True
+except ImportError:
+    HAVE_PYLIBJPEG = False
+
+try:
+    import openjpeg
+    HAVE_OPENJPEG = True
+except ImportError:
+    HAVE_OPENJPEG = False
+
+try:
+    import libjpeg
+    HAVE_LIBJPEG = True
+except ImportError:
+    HAVE_LIBJPEG = False
+
+
+from pydicom.encaps import generate_pixel_data_frame
+from pydicom.pixel_data_handlers.util import (
+    pixel_dtype, get_expected_length, reshape_pixel_array
+)
+from pydicom.uid import (
+    JPEGBaseline,
+    JPEGExtended,
+    JPEGLosslessP14,
+    JPEGLossless,
+    JPEGLSLossless,
+    JPEGLSLossy,
+    JPEG2000Lossless,
+    JPEG2000,
+    UID
+)
+
+
+LOGGER = logging.getLogger("pydicom")
+
+
+HANDLER_NAME = "pylibjpeg"
+if HAVE_PYLIBJPEG:
+    _DECODERS = get_pixel_data_decoders()
+
+_LIBJPEG_SYNTAXES = [
+    JPEGBaseline,
+    JPEGExtended,
+    JPEGLosslessP14,
+    JPEGLossless,
+    JPEGLSLossless,
+    JPEGLSLossy
+]
+_OPENJPEG_SYNTAXES = [JPEG2000Lossless, JPEG2000]
+SUPPORTED_TRANSFER_SYNTAXES = _LIBJPEG_SYNTAXES + _OPENJPEG_SYNTAXES
+
+DEPENDENCIES = {
+    "numpy": ("http://www.numpy.org/", "NumPy"),
+}
+
+
+def is_available() -> bool:
+    """Return ``True`` if the handler has its dependencies met."""
+    return HAVE_NP and HAVE_PYLIBJPEG
+
+
+def supports_transfer_syntax(tsyntax: UID) -> bool:
+    """Return ``True`` if the handler supports the `tsyntax`.
+
+    Parameters
+    ----------
+    tsyntax : pydicom.uid.UID
+        The *Transfer Syntax UID* of the *Pixel Data* that is to be used with
+        the handler.
+    """
+    return tsyntax in SUPPORTED_TRANSFER_SYNTAXES
+
+
+def needs_to_convert_to_RGB(ds: "Dataset") -> bool:
+    """Return ``True`` if the *Pixel Data* should to be converted from YCbCr to
+    RGB.
```

Comment:
With Dataset imported safely, are the quotes still needed around it?

Reply:
Because it's not. If you take out the quotes and try and `import pydicom` you'll get a bunch of `NameError` because `Dataset` only gets imported by type checkers, not when running. Using the quotes is [forward referencing](https://www.python.org/dev/peps/pep-0484/#forward-references) it (which is what I should've done with the UID typing instead of using `TypeVar`).

The same thing occurs with `np.ndarray` because there's numpy is optional. Basically using quotes means that this type will be resolved at some point later on.